src/init/features/project.ts

```diff
@@ -32,48 +34,60 @@ export interface ProjectInfo {
  */
 export async function getProjectInfo(options: any): Promise<ProjectInfo> {
   if (options.project) {
-    return selectProjectFromOptions(options);
+    return getProjectFromId(options.project);
   }
-  return selectProjectFromList(options);
+  return selectProjectInteractively();
 }
 
 /**
  * Selects project when --project is passed in.
  * @param options Command line options.
  */
-async function selectProjectFromOptions(options: any): Promise<ProjectInfo> {
+async function getProjectFromId(projectId: string): Promise<ProjectInfo> {
   let project: FirebaseProjectMetadata;
   try {
-    project = await getFirebaseProject(options.project);
+    project = await getFirebaseProject(projectId);
   } catch (e) {
-    throw new FirebaseError(`Error getting project ${options.project}: ${e}`);
+    throw new FirebaseError(
+      `Error getting project ${projectId}. Please make sure the project exists and belongs to your account.`
+    );
   }
-  const projectId = project.projectId;
-  const name = project.displayName;
-  return {
-    id: projectId,
-    label: `${projectId} (${name})`,
-    instance: _.get(project, "resources.realtimeDatabaseInstance"),
-    location: _.get(project, "resources.locationId"),
-  };
+  return toProjectInfo(project);
+}
+
+async function selectProjectInteractively(
+  pageSize: number = MAXIMUM_PROMPT_LIST
+): Promise<ProjectInfo> {
+  const { projects, nextPageToken } = await getProjectPage(pageSize);
+  if (nextPageToken) {
+    // Prompt user for project ID if we can't list all projects in 1 page
+    return selectProjectByPrompting();
+  }
+  return selectProjectFromList(projects);
+}
+
+async function selectProjectByPrompting(): Promise<ProjectInfo> {
+  const projectId = await promptOnce({
+    type: "input",
+    message: "Please input your project ID ",
+  });
+
+  return getProjectFromId(projectId);
 }
 
 /**
- * Presents user with list of projects to choose from and gets project
- * information for chosen project.
- * @param options Command line options.
+ * Presents user with list of projects to choose from and gets project information for chosen project.
  */
-async function selectProjectFromList(options: any): Promise<ProjectInfo> {
-  const projects: FirebaseProjectMetadata[] = await listFirebaseProjects();
+async function selectProjectFromList(
+  projects: FirebaseProjectMetadata[] = []
+): Promise<ProjectInfo> {
   let choices = projects.filter((p: FirebaseProjectMetadata) => !!p).map((p) => {
     return {
-      name: `${p.projectId} (${p.displayName})`,
+      name: `${p.projectId}` + (p.displayName ? ` (${p.displayName})` : ""),
```

Comment:
`` `${p.projectId}` `` can now be just `p.projectId`, since you're no longer using the special string handling.

Reply:
Done